### PR TITLE
Done adding code to handle the empty databases (fix bug)

### PR DESCRIPTION
**Pull request by brando90**
Currently there are three cases if the tables are empty:

1)Commits table is empty
2)Tests table is empty
3)test_by_score table is empty

Currently the fixed code does not crash if these are empty.

Important comment:

Commit database cannot be empty because later on code will try to do .hash on an undefined obj and it will crash. That code only executes if and only if the commits table is not empty.

Any suggestions on how this should be handled feel free to tell me! :)

Also, the importCassandra.js sript does NOT add anything to the commits table, so just running that will not be enough for me to continue working on perf. stats. Do we have a script that fills in the commits table?

### CODE REVIEW

**Comment by swong15** [comment]
Shouldn't the empty data base case be handled by 404'ing to the client on getTitle?

**Comment by brando90** [comment]
@swong15 what do you mean by getTitle? There is no such function in the code, or not in CassandraBackend.js? Do you mean getTest?

Do you mean when the tests table is empty, to throw a 404 error to the client?

**Comment by swong15** [comment]
https://github.com/gwicke/testreduce/blob/master/server.js#L167

Currently in Parsoid, a 404 error is thrown if the server doesn't have any 'work' for the client to do. Similarly if the db is empty wouldn't that correspond to the server having no work for the client to do?

**Comment by brando90** [comment]
That makes sense to me, if thats how parsoid/we want to handle it.

**Comment by gwicke** [comment]
We should definitely return a 404 when the DB is empty.

File: cql/performance_stats.cql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+--OrderPreservingPartitioner
+CREATE TABLE IF NOT EXISTS page_by_time_score (
+	commit blob,
+	pageID blob,
+	time timestamp,
```
Comment (gwicke) [inline cql/performance_stats.cql:5]: I think we want a duration here. Maybe use a float, which holds seconds?

**Comment by gwicke** [comment]
@brando90 : which bits of this patch are still needed?

**Comment by brando90** [comment]
@gwicke in my opinion all cases when the database might be empty, should be re-test to make sure everything works as expected.

**Pull request closed without merging**